### DISCUSSION TITLE
Self-hostable ?

### DISCUSSION THREAD

**Discussion by TheHellTower** [tutao/tutanota#4994] [Q&A] [answered]
Hello,

I checked a bit the building steps and seen [this](https://github.com/tutao/tutanota/blob/master/buildSrc/buildWebapp.js#L115-L126), does it mean it's not self-hostable completelly ? Is it even usable without using tutanota domain or is it just a client that connect and depend on tutanota.com itself ?

**Comment by vitoreiji** [comment, marked as answer]
We do not have a self-hostable server at the moment, but custom e-mail domain and white label domain are readily available features.

**Reply by TheHellTower** [reply]
> We do not have a self-hostable server at the moment, but custom e-mail domain and white label domain are readily available features.

I seen reddit answers from 2017 and 2020 that was saying it would be done "soon" but I suppose it will never happen then !

Thanks for your answer

**Reply by erlend-sh** [reply]
So the Tutanota email server is closed source. This took me a long time to realize, as a lot of your marketing copy describes Tutanota as an open source *service*.

Are you at liberty to disclose whether the server is completely custom or based on an open source foundation like dovecot or maddy?

**Comment by erlend-sh** [comment]
On a related note, there’s a new open source email server stack taking shape in https://stalw.art/

As you’ve already adopted Rust into your stack with @ganthern’s `mapirs`, I figured it might pique your interest 😊

**Reply by charlag** [reply]
JMAP explicitly excludes any meaningful encryption on the grounds that it's not feasible
